cmd/all-in-one/main.go

```diff
@@ -251,7 +254,13 @@ func startCollector(
 
 	{
 		r := mux.NewRouter()
-		apiHandler := collectorApp.NewAPIHandler(jaegerBatchesHandler)
+		configManager := &collectorConfigManager{samplingStrategy: strategyStore}
+		cfgParams := clientcfgHandler.HTTPHandlerParams{
+			ConfigManager:          configManager,
+			MetricsFactory:         metricsFactory,
+			LegacySamplingEndpoint: false,
+		}
+		apiHandler := collectorApp.NewAPIHandler(jaegerBatchesHandler, *clientcfgHandler.NewHTTPHandler(cfgParams))
```

Comment:
It is questionable whether all-in-one needs this, since it's already running the agent and the endpoints are available on the agent's http server. But in order to make all-in-one swappable for collector, let's keep it.

However, I don't think we need to pass cfg handler to APIHandler. We can simply call its RegisterRoutes here (same in the collector).

Nit: inline HTTPHandlerParams with NewHTTPHandler() call.